sympy/codegen/array_utils.py

```diff
@@ -1026,22 +1016,32 @@ def _codegen_array_parse(expr):
             args[i] = CodegenArrayPermuteDims(args[i], permutation)
         return CodegenArrayElementwiseAdd(*args), index0
     return expr, ()
-    raise NotImplementedError("could not recognize expression %s" % expr)
 
 
-def _parse_matrix_expression(expr):
+def parse_matrix_expression(expr: MatrixExpr) -> Basic:
+    return _parse_matrix_expression(expr)
```

Comment:
What's the purpose of this indirection? Is it just for backwards compatibility for the name `_parse_matrix_exception`?

Reply:
I just wanted to avoid exposing a recurrent function publicly. Would you prefer to change it?

Reply:
I see. Usually this is done if there is some preprocessing logic that should only be done once. I'm not sure if it's beneficial to have it done trivially like it is here.